sdk/core/core-lro/test/utils/router/paramRoutes.ts

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { createHttpHeaders, PipelineRequest, PipelineResponse } from "@azure/core-rest-pipeline";
+
+import { buildResponse, getPascalCase, parseUri } from "./utils";
+
+function putBody(request: PipelineRequest): PipelineResponse | undefined {
+  function isValidRequest(
+    initialCode: number,
+    initialState: string,
+    finalState: string,
+    finalCode: number
+  ): boolean {
+    return (
+      (initialCode === 201 &&
+        initialState === "Creating" &&
+        finalState === "Succeeded" &&
+        finalCode === 200) ||
+      (initialCode === 200 &&
+        initialState === "Updating" &&
+        finalState === "Succeeded" &&
+        finalCode === 200) ||
+      (initialCode === 201 &&
+        initialState === "Created" &&
+        finalState === "Failed" &&
+        finalCode === 200) ||
+      (initialCode === 200 &&
+        initialState === "Accepted" &&
+        finalState === "Canceled" &&
+        finalCode === 200)
```

Comment:
Are we sure these strings are going to be properly cased every time? I wonder if it would be better to always lowercase them.